dotcom-rendering/src/components/Discussion/Comments.tsx

```diff
@@ -189,6 +198,20 @@ export const Comments = ({
 		return () => clearTimeout(timer);
 	}, [expanded, comments.length, numberOfCommentsToShow, loadingMore]);
 
+	useEffect(() => {
+		if (expanded && !loadingMore && mobileDiscussionAdsEnabled) {
+			const event = new CustomEvent('comments-loaded');
+			document.dispatchEvent(event);
+		}
+	}, [comments.length, expanded, loadingMore, mobileDiscussionAdsEnabled]);
+
+	const dispatchCommentsStateChangeEvent = () => {
+		if (mobileDiscussionAdsEnabled) {
+			const event = new CustomEvent('comments-state-change');
+			document.dispatchEvent(event);
+		}
+	};
```

Comment:
I think that this method can be quite confusing, because it will only dispatch an event if the user is in the test. I believe it would be clearer if it either took `mobileDiscussionAdsEnabled` as a parameter or had the custom event live outside the component and be called conditionally:

```ts
const commentStateChangeEvent = new CustomEvent('comments-state-change');

const Comments = (props : {}) => {
  // …
  const onFilterChange = (newFilterObject: FilterOptions) => {
    // …
    mobileDiscussionAdsEnabled && document.dispatchEvent(commentStateChangeEvent);
  }
  // …
}
```

Reply:
We have refactored this Max. Thanks for the suggestion 👍🏼